sdk/framework/openapi.go

```diff
@@ -107,13 +107,12 @@ type OASLicense struct {
 }
 
 type OASPathItem struct {
-	Description       string             `json:"description,omitempty"`
-	Parameters        []OASParameter     `json:"parameters,omitempty"`
-	Sudo              bool               `json:"x-vault-sudo,omitempty" mapstructure:"x-vault-sudo"`
-	Unauthenticated   bool               `json:"x-vault-unauthenticated,omitempty" mapstructure:"x-vault-unauthenticated"`
-	CreateSupported   bool               `json:"x-vault-createSupported,omitempty" mapstructure:"x-vault-createSupported"`
-	DisplayNavigation bool               `json:"x-vault-displayNavigation,omitempty" mapstructure:"x-vault-displayNavigation"`
-	DisplayAttrs      *DisplayAttributes `json:"x-vault-displayAttrs,omitempty" mapstructure:"x-vault-displayAttrs"`
+	Description     string             `json:"description,omitempty"`
+	Parameters      []OASParameter     `json:"parameters,omitempty"`
+	Sudo            bool               `json:"x-vault-sudo,omitempty" mapstructure:"x-vault-sudo"`
+	Unauthenticated bool               `json:"x-vault-unauthenticated,omitempty" mapstructure:"x-vault-unauthenticated"`
+	CreateSupported bool               `json:"x-vault-createSupported,omitempty" mapstructure:"x-vault-createSupported"`
+	DisplayAttrs    *DisplayAttributes `json:"x-vault-displayAttrs,omitempty" mapstructure:"x-vault-displayAttrs"`
```

Comment:
Note to reviewer: `DisplayNavigation` removed as wholly unused.

Reply:
🤔 If I understand correctly, removing a public field of a public struct in our SDK is a breaking change in terms of binary compatibility and so breaks SemVer and therefore Go mod's rules.

I don't know the ecosystem well enough to estimate whether this would actually impact any real downstream code, but for example it could cause downstream code to fail to build if they are referring to this field.

I'd be happy to be advised by any one who knows the code better whether there is some exceptional circumstance that makes it seem OK in this case, but my instinct would be to not remove this and avoid the issue entirely. I certainly don't think it's worth making a v2 of the whole sdk module to change this but that would be the "correct" way to do it I think.

Am I missing some reason this is fine?

Reply:
You are correct it is a compatibility break. (Though given that Go doesn't generally ship libraries as binaries, but as source code, it's more a source compatibility issue than a binary one.)

However, the version of the SDK is v0.x.x, and the Go versioning rules explicitly declare that version zero is way of declaring a module has yet to implement any compatibility guarantees as it is still in development. Therefore, by keeping the SDK version at v0.x.x, the implicit message is that rights are reserved to break compatibility as desired.

(In contrast, the API package is v1.x.x implying compatibility will be maintained.)

I'd need to do a quick bit of research to confirm, but I think the SDK's lack of compatibility guarantees has been exercised in the past - e.g. in recent changes to decouple the Vault version number from the SDK.

Practically speaking, the field is located in a type which is only used to perform JSON serialization of OpenAPI specifications, internally to the SDK. It is very likely that it has zero uses in the wider ecosystem.

Therefore whilst it is a compatibility break, there are both technical and pragmatic reasons to believe that fine to do so.

**However** it is only a bit of cleanup in passing, and I would be happy to remove that change from this PR, if you prefer it that way.

Reply:
I agree with @maxb on the rationale behind breaking the backwards compatibility and I think it's OK to remove this 👍

Reply:
Just to indulge my curiosity about the compatibility issue, I ran previous versions of the SDK through an API diff tool:

# Incompatible changes between v0.2.0 and v0.2.1

- github.com/hashicorp/vault/sdk/framework/TypeBool: value changed from 3 to 4
- github.com/hashicorp/vault/sdk/framework/TypeCommaIntSlice: value changed from 13 to 14
- github.com/hashicorp/vault/sdk/framework/TypeCommaStringSlice: value changed from 9 to 10
- github.com/hashicorp/vault/sdk/framework/TypeDurationSecond: value changed from 5 to 6
- github.com/hashicorp/vault/sdk/framework/TypeFloat: value changed from 15 to 16
- github.com/hashicorp/vault/sdk/framework/TypeHeader: value changed from 14 to 15
- github.com/hashicorp/vault/sdk/framework/TypeKVPairs: value changed from 12 to 13
- github.com/hashicorp/vault/sdk/framework/TypeLowerCaseString: value changed from 10 to 11
- github.com/hashicorp/vault/sdk/framework/TypeMap: value changed from 4 to 5
- github.com/hashicorp/vault/sdk/framework/TypeNameString: value changed from 11 to 12
- github.com/hashicorp/vault/sdk/framework/TypeSignedDurationSecond: value changed from 6 to 7
- github.com/hashicorp/vault/sdk/framework/TypeSlice: value changed from 7 to 8
- github.com/hashicorp/vault/sdk/framework/TypeStringSlice: value changed from 8 to 9
- github.com/hashicorp/vault/sdk/framework/TypeTime: value changed from 16 to 17

# Incompatible changes between v0.2.1 and v0.3.0

- github.com/hashicorp/vault/sdk/logical/HTTPRawCacheControl: removed
- github.com/hashicorp/vault/sdk/plugin/pb/BackendServer.mustEmbedUnimplementedBackendServer: added unexported method
- github.com/hashicorp/vault/sdk/plugin/pb/ProtoConnectionToLogicalConnection: changed from func(*Connection) *github.com/hashicorp/vault/sdk/logical.Connection to func(*Connection) (*github.com/hashicorp/vault/sdk/logical.Connection, error)
- github.com/hashicorp/vault/sdk/plugin/pb/RegisterBackendServer: changed from func(*google.golang.org/grpc.Server, BackendServer) to func(google.golang.org/grpc.ServiceRegistrar, BackendServer)
- github.com/hashicorp/vault/sdk/plugin/pb/RegisterStorageServer: changed from func(*google.golang.org/grpc.Server, StorageServer) to func(google.golang.org/grpc.ServiceRegistrar, StorageServer)
- github.com/hashicorp/vault/sdk/plugin/pb/RegisterSystemViewServer: changed from func(*google.golang.org/grpc.Server, SystemViewServer) to func(google.golang.org/grpc.ServiceRegistrar, SystemViewServer)
- github.com/hashicorp/vault/sdk/plugin/pb/StorageServer.mustEmbedUnimplementedStorageServer: added unexported method
- github.com/hashicorp/vault/sdk/plugin/pb/SystemViewServer.mustEmbedUnimplementedSystemViewServer: added unexported method
- github.com/hashicorp/vault/sdk/database/dbplugin/DatabaseServer.mustEmbedUnimplementedDatabaseServer: added unexported method
- github.com/hashicorp/vault/sdk/database/dbplugin/RegisterDatabaseServer: changed from func(*google.golang.org/grpc.Server, DatabaseServer) to func(google.golang.org/grpc.ServiceRegistrar, DatabaseServer)
- github.com/hashicorp/vault/sdk/database/dbplugin/v5/proto/DatabaseServer.mustEmbedUnimplementedDatabaseServer: added unexported method
- github.com/hashicorp/vault/sdk/database/dbplugin/v5/proto/RegisterDatabaseServer: changed from func(*google.golang.org/grpc.Server, DatabaseServer) to func(google.golang.org/grpc.ServiceRegistrar, DatabaseServer)
- package github.com/hashicorp/vault/sdk/testing/stepwise: removed
- package github.com/hashicorp/vault/sdk/testing/stepwise/environments/docker: removed
- package github.com/hashicorp/vault/sdk/helper/awsutil: removed

# Incompatible changes between v0.3.0 and v0.4.0

- github.com/hashicorp/vault/sdk/logical/SystemView.NewPluginClient: added
- github.com/hashicorp/vault/sdk/helper/pluginutil/RunnerUtil.NewPluginClient: added
- github.com/hashicorp/vault/sdk/helper/certutil/ValidateSignatureLength: changed from func(int) error to func(string, int) error
- github.com/hashicorp/vault/sdk/database/dbplugin/v5/DatabasePluginClient.Mutex: removed
- github.com/hashicorp/vault/sdk/database/dbplugin/v5/GRPCDatabasePlugin: old is comparable, new is not
- github.com/hashicorp/vault/sdk/database/dbplugin/v5/NewPluginClient: changed from func(context.Context, github.com/hashicorp/vault/sdk/helper/pluginutil.RunnerUtil, *github.com/hashicorp/vault/sdk/helper/pluginutil.PluginRunner, github.com/hashicorp/go-hclog.Logger, bool) (Database, error) to func(context.Context, github.com/hashicorp/vault/sdk/helper/pluginutil.RunnerUtil, github.com/hashicorp/vault/sdk/helper/pluginutil.PluginClientConfig) (Database, error)
- sync/(*Mutex).Lock, method set of *DatabasePluginClient: removed
- sync/(*Mutex).TryLock, method set of *DatabasePluginClient: removed
- sync/(*Mutex).Unlock, method set of *DatabasePluginClient: removed

# Incompatible changes between v0.4.0 and v0.4.1


# Incompatible changes between v0.4.1 and v0.5.0

- github.com/hashicorp/vault/sdk/helper/useragent/PluginString: changed from func(*github.com/hashicorp/vault/sdk/logical.PluginEnvironment, string) string to func(*github.com/hashicorp/vault/sdk/logical.PluginEnvironment, string, ...string) string
- github.com/hashicorp/vault/sdk/helper/useragent/String: changed from func() string to func(...string) string

# Incompatible changes between v0.5.0 and v0.5.1


# Incompatible changes between v0.5.1 and v0.5.2

- github.com/hashicorp/vault/sdk/helper/keysutil/ParsePKCS8Ed25519PrivateKey: removed

# Incompatible changes between v0.5.2 and v0.5.3


# Incompatible changes between v0.5.3 and v0.6.0

- github.com/hashicorp/vault/sdk/database/dbplugin/PluginFactory: changed from func(context.Context, string, github.com/hashicorp/vault/sdk/helper/pluginutil.LookRunnerUtil, github.com/hashicorp/go-hclog.Logger) (Database, error) to func(context.Context, string, string, github.com/hashicorp/vault/sdk/helper/pluginutil.LookRunnerUtil, github.com/hashicorp/go-hclog.Logger) (Database, error)
- github.com/hashicorp/vault/sdk/helper/pluginutil/LookRunnerUtil: changed from interface{github.com/hashicorp/vault/sdk/helper/pluginutil.Looker; github.com/hashicorp/vault/sdk/helper/pluginutil.RunnerUtil} to string
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedEncryptingKeyByName: added
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedEncryptingKeyByUUID: added
- github.com/hashicorp/vault/sdk/logical/SystemView.ListVersionedPlugins: added
- github.com/hashicorp/vault/sdk/logical/SystemView.LookupPluginVersion: added
- github.com/hashicorp/vault/sdk/helper/pluginutil/Looker.LookupPluginVersion: added
- github.com/hashicorp/vault/sdk/helper/pluginutil/MultiplexingSupported: changed from func(context.Context, google.golang.org/grpc.ClientConnInterface) (bool, error) to func(context.Context, google.golang.org/grpc.ClientConnInterface, string) (bool, error)
- github.com/hashicorp/vault/sdk/helper/pluginutil/PluginCACertPEMEnv: changed from var to const
- github.com/hashicorp/vault/sdk/helper/pluginutil/PluginClient.Reload: added
- github.com/hashicorp/vault/sdk/helper/pluginutil/PluginMetadataModeEnv: changed from var to const
- github.com/hashicorp/vault/sdk/helper/pluginutil/PluginMlockEnabled: changed from var to const
- github.com/hashicorp/vault/sdk/helper/pluginutil/PluginUnwrapTokenEnv: changed from var to const
- github.com/hashicorp/vault/sdk/helper/pluginutil/PluginVaultVersionEnv: changed from var to const
- github.com/hashicorp/vault/sdk/plugin/BackendPluginClient.Mutex: removed
- sync/(*Mutex).Lock, method set of *BackendPluginClient: removed
- sync/(*Mutex).TryLock, method set of *BackendPluginClient: removed
- sync/(*Mutex).Unlock, method set of *BackendPluginClient: removed

# Incompatible changes between v0.6.0 and v0.6.1

- github.com/hashicorp/vault/sdk/helper/keysutil/(*Policy).EncryptWithFactory: changed from func(int, []byte, []byte, string, AEADFactory) (string, error) to func(int, []byte, []byte, string, ...interface{}) (string, error)

# Incompatible changes between v0.6.1 and v0.6.2

- github.com/hashicorp/vault/sdk/helper/pluginutil/RunnerUtil.VaultVersion: added
- github.com/hashicorp/vault/sdk/framework/NewOASDocument: changed from func() *OASDocument to func(string) *OASDocument
- github.com/hashicorp/vault/sdk/framework/Response: old is comparable, new is not
- github.com/hashicorp/vault/sdk/helper/consts/VaultAllowPendingRemovalMountsEnv: removed
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedMACKeyByName: added
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedMACKeyByUUID: added
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.VaultVersion: removed
- github.com/hashicorp/vault/sdk/logical/SystemView.VaultVersion: added

# Incompatible changes between v0.6.2 and v0.7.0


# Incompatible changes between v0.7.0 and v0.8.0

- github.com/hashicorp/vault/sdk/helper/keysutil/Policy.ManagedKeyName: removed
- github.com/hashicorp/vault/sdk/plugin/pb/SystemViewClient.ClusterInfo: added
- package github.com/hashicorp/vault/sdk/version: removed
- context/Context: changed from interface{Deadline() (deadline time.Time, ok bool); Done() <-chan struct{}; Err() error; Value(key any) any} to int
- github.com/hashicorp/vault/sdk/logical/(*StorageView).Delete: changed from func(context.Context, string) error to func(context.Context, string) error
- github.com/hashicorp/vault/sdk/logical/(*StorageView).Get: changed from func(context.Context, string) (*StorageEntry, error) to func(context.Context, string) (*StorageEntry, error)
- github.com/hashicorp/vault/sdk/logical/(*StorageView).List: changed from func(context.Context, string) ([]string, error) to func(context.Context, string) ([]string, error)
- github.com/hashicorp/vault/sdk/logical/(*StorageView).Put: changed from func(context.Context, *StorageEntry) error to func(context.Context, *StorageEntry) error
- github.com/hashicorp/vault/sdk/logical/ManagedAsymmetricKey.GetPublicKey: changed from func(context.Context) (crypto.PublicKey, error) to func(context.Context) (crypto.PublicKey, error)
- github.com/hashicorp/vault/sdk/logical/ManagedEncryptingKey.Decrypt: added
- github.com/hashicorp/vault/sdk/logical/ManagedEncryptingKey.Encrypt: added
- github.com/hashicorp/vault/sdk/logical/ManagedEncryptingKey.GetAEAD: removed
- github.com/hashicorp/vault/sdk/logical/ManagedKey.Present: changed from func(context.Context) (bool, error) to func(context.Context) (bool, error)
- github.com/hashicorp/vault/sdk/logical/ManagedKeyRandomSource.GetRandomBytes: changed from func(context.Context, int) ([]byte, error) to func(int) ([]byte, error)
- github.com/hashicorp/vault/sdk/logical/ManagedKeyRandomSourceConsumer: changed from func(context.Context, ManagedKeyRandomSource) error to func(context.Context, ManagedKeyRandomSource) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedEncryptingKeyByName: changed from func(context.Context, string, string, ManagedEncryptingKeyConsumer) error to func(context.Context, string, string, ManagedEncryptingKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedEncryptingKeyByUUID: changed from func(context.Context, string, string, ManagedEncryptingKeyConsumer) error to func(context.Context, string, string, ManagedEncryptingKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedKeyByName: changed from func(context.Context, string, string, ManagedKeyConsumer) error to func(context.Context, string, string, ManagedKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedKeyByUUID: changed from func(context.Context, string, string, ManagedKeyConsumer) error to func(context.Context, string, string, ManagedKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedMACKeyByName: changed from func(context.Context, string, string, ManagedMACKeyConsumer) error to func(context.Context, string, string, ManagedMACKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedMACKeyByUUID: changed from func(context.Context, string, string, ManagedMACKeyConsumer) error to func(context.Context, string, string, ManagedMACKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedSigningKeyByName: changed from func(context.Context, string, string, ManagedSigningKeyConsumer) error to func(context.Context, string, string, ManagedSigningKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedKeySystemView.WithManagedSigningKeyByUUID: changed from func(context.Context, string, string, ManagedSigningKeyConsumer) error to func(context.Context, string, string, ManagedSigningKeyConsumer) error
- github.com/hashicorp/vault/sdk/logical/ManagedMACKey.MAC: changed from func(context.Context, string, []byte) ([]byte, error) to func(context.Context, string, []byte) ([]byte, error)
- github.com/hashicorp/vault/sdk/logical/ManagedMACKeyConsumer: changed from func(context.Context, ManagedMACKey) error to func(context.Context, ManagedMACKey) error
- github.com/hashicorp/vault/sdk/logical/ManagedSigningKey.GetSigner: changed from func(context.Context) (crypto.Signer, error) to func(context.Context) (crypto.Signer, error)
- github.com/hashicorp/vault/sdk/logical/ManagedSigningKey.Sign: changed from func(context.Context, []byte, io.Reader, crypto.SignerOpts) ([]byte, error) to func(context.Context, []byte, io.Reader, crypto.SignerOpts) ([]byte, error)
- github.com/hashicorp/vault/sdk/logical/ManagedSigningKey.Verify: changed from func(context.Context, []byte, []byte, crypto.SignerOpts) (bool, error) to func(context.Context, []byte, []byte, crypto.SignerOpts) (bool, error)
- github.com/hashicorp/vault/sdk/logical/ManagedSigningKeyConsumer: changed from func(context.Context, ManagedSigningKey) error to func(context.Context, ManagedSigningKey) error
- github.com/hashicorp/vault/sdk/logical/PasswordPolicy.Generate: changed from func(context.Context, io.Reader) (string, error) to func(context.Context, io.Reader) (string, error)
- github.com/hashicorp/vault/sdk/logical/PluginVersionClient.Version: changed from func(context.Context, *Empty, ...google.golang.org/grpc.CallOption) (*VersionReply, error) to func(context.Context, *Empty, ...google.golang.org/grpc.CallOption) (*VersionReply, error)
- github.com/hashicorp/vault/sdk/logical/PluginVersionServer.Version: changed from func(context.Context, *Empty) (*VersionReply, error) to func(context.Context, *Empty) (*VersionReply, error)
- github.com/hashicorp/vault/sdk/logical/ScanView: changed from func(context.Context, ClearableView, func(path string)) error to func(context.Context, ClearableView, func(path string)) error
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.ForwardGenericRequest: changed from func(context.Context, *Request) (*Response, error) to func(context.Context, *Request) (*Response, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.GeneratePasswordFromPolicy: changed from func(context.Context, string) (string, error) to func(context.Context, string) (string, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.ListVersionedPlugins: changed from func(context.Context, github.com/hashicorp/vault/sdk/helper/consts.PluginType) ([]github.com/hashicorp/vault/sdk/helper/pluginutil.VersionedPlugin, error) to func(context.Context, github.com/hashicorp/vault/sdk/helper/consts.PluginType) ([]github.com/hashicorp/vault/sdk/helper/pluginutil.VersionedPlugin, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.LookupPlugin: changed from func(context.Context, string, github.com/hashicorp/vault/sdk/helper/consts.PluginType) (*github.com/hashicorp/vault/sdk/helper/pluginutil.PluginRunner, error) to func(context.Context, string, github.com/hashicorp/vault/sdk/helper/consts.PluginType) (*github.com/hashicorp/vault/sdk/helper/pluginutil.PluginRunner, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.LookupPluginVersion: changed from func(context.Context, string, github.com/hashicorp/vault/sdk/helper/consts.PluginType, string) (*github.com/hashicorp/vault/sdk/helper/pluginutil.PluginRunner, error) to func(context.Context, string, github.com/hashicorp/vault/sdk/helper/consts.PluginType, string) (*github.com/hashicorp/vault/sdk/helper/pluginutil.PluginRunner, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.NewPluginClient: changed from func(context.Context, github.com/hashicorp/vault/sdk/helper/pluginutil.PluginClientConfig) (github.com/hashicorp/vault/sdk/helper/pluginutil.PluginClient, error) to func(context.Context, github.com/hashicorp/vault/sdk/helper/pluginutil.PluginClientConfig) (github.com/hashicorp/vault/sdk/helper/pluginutil.PluginClient, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.PluginEnv: changed from func(context.Context) (*PluginEnvironment, error) to func(context.Context) (*PluginEnvironment, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.ResponseWrapData: changed from func(context.Context, map[string]interface{}, time.Duration, bool) (*github.com/hashicorp/vault/sdk/helper/wrapping.ResponseWrapInfo, error) to func(context.Context, map[string]interface{}, time.Duration, bool) (*github.com/hashicorp/vault/sdk/helper/wrapping.ResponseWrapInfo, error)
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.SudoPrivilege: changed from func(context.Context, string, string) bool to func(context.Context, string, string) bool
- github.com/hashicorp/vault/sdk/logical/StaticSystemView.VaultVersion: changed from func(context.Context) (string, error) to func(context.Context) (string, error)
- github.com/hashicorp/vault/sdk/logical/SystemView.ClusterID: added
- github.com/hashicorp/vault/sdk/logical/UnimplementedPluginVersionServer.Version: changed from func(context.Context, *Empty) (*VersionReply, error) to func(context.Context, *Empty) (*VersionReply, error)

# Incompatible changes between v0.8.0 and v0.8.1

- github.com/hashicorp/vault/sdk/logical/(*EventData).GetMetadata: changed from func() []byte to func() *google.golang.org/protobuf/types/known/structpb.Struct
- github.com/hashicorp/vault/sdk/logical/EventData.Metadata: changed from []byte to *google.golang.org/protobuf/types/known/structpb.Struct

# Incompatible changes between v0.8.1 and v0.9.0

- github.com/hashicorp/vault/sdk/logical/(*EventData).ID: removed
- github.com/hashicorp/vault/sdk/logical/(*EventReceived).GetTimestamp: removed
- github.com/hashicorp/vault/sdk/logical/EventReceived.Timestamp: removed

# Incompatible changes between v0.9.0 and v0.9.1

- github.com/hashicorp/vault/sdk/plugin/(*BackendPluginClient).IsExternal: removed
- github.com/hashicorp/vault/sdk/logical/Externaler: removed
- github.com/hashicorp/vault/sdk/helper/ldaputil/LDAP.Dial: removed
- github.com/hashicorp/vault/sdk/helper/ldaputil/LDAP.DialTLS: removed
- github.com/hashicorp/vault/sdk/helper/ldaputil/LDAP.DialURL: added

# Incompatible changes between v0.9.1 and HEAD

- github.com/hashicorp/vault/sdk/framework/OASPathItem.DisplayNavigation: removed
- github.com/hashicorp/vault/sdk/helper/testcluster/WaitForReplicationStatus: changed from func(context.Context, *github.com/hashicorp/vault/api.Client, bool, func(map[string]interface{}) bool) error to func(context.Context, *github.com/hashicorp/vault/api.Client, bool, func(map[string]interface{}) error) error

Reply:
It turns out the supposedly compatible API has fudged things a bit in the past, too:

# Incompatible changes between v1.0.4 and v1.1.0

- github.com/hashicorp/vault/api/(*Sys).ConfigureCORS: changed from func(*CORSRequest) (*CORSResponse, error) to func(*CORSRequest) error
- github.com/hashicorp/vault/api/(*Sys).DisableCORS: changed from func() (*CORSResponse, error) to func() error
- github.com/hashicorp/vault/api/CORSRequest.AllowedOrigins: changed from string to []string
- github.com/hashicorp/vault/api/CORSRequest: old is comparable, new is not
- github.com/hashicorp/vault/api/CORSResponse.AllowedOrigins: changed from string to []string
- github.com/hashicorp/vault/api/CORSResponse: old is comparable, new is not

# Incompatible changes between v1.5.0 and v1.6.0

- github.com/hashicorp/vault/api/(*OutputStringError).CurlString: changed from func() string to func() (string, error)
- github.com/hashicorp/vault/api/(*Sys).Monitor: changed from func(context.Context, string) (chan string, error) to func(context.Context, string, string) (chan string, error)
- github.com/hashicorp/vault/api/AutopilotServer.Meta: removed
- github.com/hashicorp/vault/api/TLSConfig: old is comparable, new is not

# Incompatible changes between v1.7.2 and v1.8.0

- github.com/hashicorp/vault/api/PluginMetadataModeEnv: changed from var to const
- github.com/hashicorp/vault/api/PluginUnwrapTokenEnv: changed from var to const

# Incompatible changes between v1.8.0 and v1.8.1

- github.com/hashicorp/vault/api/MountInput.PluginVersion: removed

# Incompatible changes between v1.8.2 and v1.8.3

- github.com/hashicorp/vault/api/SealStatusResponse: old is comparable, new is not

# Incompatible changes between v1.9.0 and v1.9.1

- github.com/hashicorp/vault/api/OutputPolicyError: old is comparable, new is not